features/promotion/complex_promotions.feature

```diff
@@ -0,0 +1,68 @@
+@promotion
+Feature: Receiving a discount based on a configured promotion
+    In order to pay less for my order during a promotion period
+    As a Customer
+    I want to receive a discount for my purchase
+
+    Background:
+        Given the store operates on a single channel in "France"
+        And the store classifies its products as "Jackets", "Trousers", "Formal attire" and "Dresses"
+        And the store has a product "Black Sabbath jacket" priced at "€100.00"
+        And this product belongs to "Jackets"
+        And the store has a product "Iron Maiden trousers" priced at "€80.00"
+        And this product belongs to "Trousers"
+        And the store has a product "Metallica dress" priced at "€50.00"
+        And this product belongs to "Dresses"
+        And the store has a product "Rammstein bow tie" priced at "€10.00"
+        And this product belongs to "Formal attire"
+        And the store has "DHL" shipping method with "€10.00" fee
+
+    @ui
+    Scenario: Receiving a discount on the first order
+        Given there is a promotion "First order promotion"
+        And it gives "20%" off on the customer's 1st order
+        And I am logged in customer
+        When I add product "Metallica Dress" to the cart
+        Then my cart total should be "€40.00"
+        And my discount should be "-€10.00"
+
+    @ui
+    Scenario: Receiving a discount on products from a specific taxon if an order contains products from an another taxon
+        Given there is a promotion "Jacket-trousers pack"
+        And it gives "10%" off on every product classified as "Jackets" if an order contains any product classified as "Trousers"
+        When I add product "Iron Maiden trousers" to the cart
+        And I add product "Black Sabbath jacket" to the cart
+        Then product "Black Sabbath jacket" price should be decreased by "€10.00"
+        And my cart total should be "€170.00"
+        And my discount should be "-€10.00"
+
+    @ui
+    Scenario: Receiving a discount on items and shipping from one promotion based on items total
+        Given there is a promotion "Jackets and shipping discount"
+        And it gives "€10.00" off on every product classified as "Jackets" and a free shipping to every order with items total equal at least "€500.00"
+        And I am logged in customer
+        When I add 7 products "Black Sabbath jacket" to the cart
+        And I proceed selecting "DHL" shipping method
+        Then theirs price should be decreased by "€70.00"
+        And my cart total should be "€630.00"
+        And my discount should be "-€80.00"
+        And my cart shipping fee should be "€10.00"
```

Comment:
On the cart summary page, shipping fee is always displayed but it's discount is counted in "Promotions total" row. Should we cross out an old shipping fee and display new below, as in discounted item row?